interface/forms/sdoh/new.php

```diff
@@ -156,6 +184,25 @@ function CalculateTotal() {
             document.getElementById('totalscorerender').innerHTML = totalscore;
             document.getElementById('totalscore').value = totalscore;
         }
+
+        <?php if ($patientPortalSession || $patientPortalOther) { ?>
+            // code to support form in the patient portal
+            $(function () {
+                window.addEventListener("message", (e) => {
+                    if (event.origin !== window.location.origin) {
+                        return false;
+                    }
+                    if (e.data.submitForm === true) {
+                        e.preventDefault();
+                        document.forms[0].submit();
+                    }
+                    <?php if (($mode == "update") && $patientPortalOther) { ?>
+                        parent.postMessage({formid:<?php echo js_escape($_GET['id']); ?>}, window.location.origin);
+                    <?php } ?>
+                });
+            });
```

Comment:
Glad you caught this. 
Although, all these modifications that are necessary to make an encounter form compliant with portal seems to me cumbersome for hope others can follow the pattern.

I've been thinking about trying to come up with a canned bootstrap in forms for doing the same.